cmd/clusterctl/pkg/internal/test/fake_proxy.go

```diff
@@ -142,3 +144,27 @@ func (f *FakeProxy) WithProviderInventory(name string, providerType clusterctlv1
 
 	return f
 }
+
+// nameToProviderLabelName returns the ProviderLabelName for a provider.
+// Nb. this function mirrors util.NameToProviderLabelName that can't be used here to avoid circular dependencies
+func nameToProviderLabelName(name string, providerType clusterctlv1.ProviderType) string {
+	switch providerType {
+	case clusterctlv1.CoreProviderType:
+		return name
+	case clusterctlv1.BootstrapProviderType:
+		if name == "kubeadm-bootstrap" {
```

Comment:
Why do we need to do this instead of performing validation?

Reply:
changed approach; does not apply anymore